### DISCUSSION TITLE
Deconstruction to local variable

### DISCUSSION THREAD

**Discussion by julre** [dotnet/csharplang#5869] [Language Ideas]
### Background

When you need to deconstruct multiple objects at the same scope naming could be hard.

``` cs

class Person 
{
    public string Name { get; set; }
    public int Age { get; set; }
    // Some other properties
} 

class Group
{
    public string Name { get; set; }
    public string Description { get; set; }
    // Some other properties
} 

(string personName, int personAge) = personObj;
(string groupName, _) = groupObj;

Console.WriteLine($"{groupName}: {personName} {personAge}")

```

### Proposal

The abillity the deconstruct the object to an local variable

``` cs

(string Name, int Age) personSimple = personObj;
(string Name, _) groupSimple = groupObj;

Console.WriteLine($"{groupSimple.Name}: {personSimple.Name} {personSimple.Age}")

```

**Comment by svick** [comment]
I'm not sure I understand why you would want this. Why do you want to access `personSimple.Name` instead of `personObj.Name`?

**Comment by HaloFour** [comment]
This proposal seems to assume that deconstruction is nominal or produces a new type with the shape of the suggested tuple, and neither are the case.  Deconstruction is purely positional, the variable names are (intentionally) meaningless and there is no other container.

For reference:

```cs
(string personName, int personAge) = personObj;
```

translates into:

```cs
string personName;
int personAge;
personObj.Deconstruct(out personName, out personInt);
```
